lib/commands/debug.js

```diff
@@ -0,0 +1,180 @@
+var command = {
+  command: 'debug',
+  description: 'Interactively debug any transaction on the blockchain (experimental)',
+  builder: {
+    _: {
+      type: "string"
+    }
+  },
+  run: function (options, done) {
+    var Config = require("truffle-config");
+    var Debugger = require("truffle-debugger");
+    var Environment = require("../environment");
+    var repl = require("repl");
+    var OS = require("os");
+
+    var config = Config.detect(options);
+
+    Environment.detect(config, function(err) {
+      if (err) return done(err);
+
+      if (config._.length == 0) {
+        return done(new Error("Please specify a transaction hash as the first parameter in order to debug that transaction. i.e., truffle debug 0x1234..."));
+      }
+
+      config.logger.log("The interactive debugger is a proof of concept and is only intended for use with Ganache." + OS.EOL + "Please report any issues to the Truffle dev team.");
+      config.logger.log("");
+
+      var tx_hash = config._[0];
+
+      var bugger = new Debugger(config);
+
+      bugger.start(tx_hash, function(err) {
+        if (err) return done(err);
+
+        var help = "Commands:" + OS.EOL + "(enter) step over, (i) step into, (o) step out, (n) step next, (;) step instruction" + OS.EOL + "(p) print instruction, (s) print stack information, (h) print this help, (q) quit";
```

Comment:
Would definitely be a good feature. We can consider it in another pass (though I suspect Ganache is where all the good debugging UI is going to go).